extensions/sd_api_pictures/script.py

```diff
@@ -97,19 +152,51 @@ def input_modifier(string):
 
     global params
 
-    if not params['mode'] == 1:  # if not in immersive/interactive mode, do nothing
-        return string
-
-    if triggers_are_in(string):  # if we're in it, check for trigger words
+    if params['mode'] == 1:  # if not in immersive/interactive mode, do nothing
+            return if_of_is_in(string)
+    if params['mode'] == 2:
+        params['characterfocus'] = False
         toggle_generation(True)
         string = string.lower()
-        if "of" in string:
-            subject = string.split('of', 1)[1]  # subdivide the string once by the first 'of' instance and get what's coming after it
-            string = "Please provide a detailed and vivid description of " + subject
-        else:
-            string = "Please provide a detailed description of your appearance, your surroundings and what you are doing right now"
+        return string
+    if params['mode'] == 0:
+        toggle_generation(False)
```

Comment:
You shouldn't toggle the generation here at all because the current state could be FORCED (user has pressed the 'Force pic' button while in Manual mode before submitting the text)